kernel/rros/clock.rs

```diff
@@ -586,13 +587,91 @@ fn set_timer_value(timer: Arc<SpinLock<RrosTimer>>, value: &Itimerspec64) -> Res
     return Ok(0);
 }
 
+pub fn rros_current_rq() -> *mut rros_rq {
+    let current = rros_current();
+    unsafe { (*(*current).locked_data().get()).rq.unwrap() }
+}
+
+pub fn double_timer_base_lock(tb1: *mut RrosTimerbase, tb2: *mut RrosTimerbase) {
+    unsafe {
+        if tb1 == tb2 {
+            (*tb1).lock.raw_spin_lock();
+        } else if tb1 < tb2 {
+            (*tb1).lock.raw_spin_lock();
+            (*tb2).lock.raw_spin_lock_nested(bindings::SINGLE_DEPTH_NESTING);
+        } else {
+            (*tb2).lock.raw_spin_lock();
+            (*tb1).lock.raw_spin_lock_nested(bindings::SINGLE_DEPTH_NESTING);
+        }    
+    }
+}
+
+pub fn double_timer_base_unlock(tb1: *mut RrosTimerbase, tb2: *mut RrosTimerbase) {
+    unsafe {
+        (*tb1).lock.raw_spin_unlock();
+        if tb1 != tb2 {
+            (*tb2).lock.raw_spin_unlock();
+        }
+    }
+}
+
+pub fn rros_move_timer(timer: Arc<SpinLock<RrosTimer>>, clock: *mut RrosClock, mut rq: *mut rros_rq) {
```

Comment:
The clocks are globally static reference. Maybe we can use the reference to avoid so many pointers. If you use the pointer to avoid multiple mutable reference rules in Rust, you may encounter the concurrent issue if you just use unsafe. You can try Mutex or something else to get inner mutability. However, as the last resort, try an unsafe cell rather than use a raw pointer.

Reply:
There are many cases of using raw pointers in our code for this reason, such as `RrosTimerbase` and `rros_rq`. I think we need a new `PR` to solve this common problem.

Reply:
I will add a **TODO:** in the code.